include/RandLAPACK/comps/util.hh

```diff
@@ -5,69 +5,104 @@
 
 namespace RandLAPACK::comps::util {
 
-template <typename T>
-void eye(
-        int64_t m,
-        int64_t n,
-        std::vector<T>& A
-);
-
 /*
-Extracts the l-portion of the GETRF result, places 1's on the main diagonal.
+Generates an identity matrix.
 */
-template <typename T> 
-void get_L(
-        int64_t m,
-        int64_t n,
-        std::vector<T>& L
+template <typename T>
+void eye(
+    int64_t m,
+    int64_t n,
+    std::vector<T>& A
 );
 
 /*
 Overwrites the diagonal entries of matrix S with those stored in s.
 */
 template <typename T>
 void diag(
-        int64_t m,
-        int64_t n,
-        const std::vector<T>& s,
-        int64_t k,
-        std::vector<T>& S
+    int64_t m,
+    int64_t n,
+    const std::vector<T>& s,
+    int64_t k,
+    std::vector<T>& S
 );
 
+/*
+Captyres k diagonal elements of A and stores them in buf.
+*/
+template <typename T> 
+void extract_diag(
+    int64_t m,
+    int64_t n,
+    int64_t k, 
+    std::vector<T>& A,
```

Comment:
pass by ref to const

Reply:
Fixed in entire file, where appropriate.